tests/HashMapProperties.hs

```diff
@@ -49,6 +49,14 @@ pFoldable :: [(Int, Int)] -> Bool
 pFoldable = (L.sort . Foldable.foldr (:) []) `eq`
             (L.sort . Foldable.foldr (:) [])
 
+pHashable :: [(Key, Int)] -> [Int] -> Int -> Property
+pHashable xs is salt =
+    x == y ==> hashWithSalt salt x === hashWithSalt salt y
```

Comment:
Lets see if I can paraphrase the test to see if I understand what it's doing. The test is

 * checking that two equal hash maps hash to the same value, where
 * given that they were created from the same list (which was permuted in one case).

Correct?

Given the way we generate the input list, will we end up throwing away most of the generated lists (by using `==>`) because they contain duplicate keys but different values? I guess that's fine as long as QC doesn't give up generating input (which it does if it can't find 100 cases after generating 500, I believe.)